### PR TITLE
feature/shared file systems: create share

### PR DESCRIPTION
**Pull request by mvalkon**
Adds support for creating a [`share`][1]. 

For #129  

[1]: http://developer.openstack.org/api-ref/shared-file-systems/?expanded=create-share-detail

### CODE REVIEW

**Comment by mvalkon** [comment]
If possible, I'd like to add the acceptance tests in a separate commit, hopefully after #118 is merged to upstream master as it sets up some infrastructure.

**Comment by mvalkon** [comment]
Some of the share creation source can be found here:

https://github.com/openstack/manila/blob/16c522bf5e5c3d37018caef11dd7852462193327/manila/share/api.py#L221

**Comment by mvalkon** [comment]
@jrperritt please review when you have time

**Comment by mvalkon** [comment]
Any pointers on ways to orchestrate the PR's would be valued. We're aiming to produce small PR's implementing one API operation each, but this leads to PR's having duplicated content until merged.

**Comment by mvalkon** [comment]
@jrperritt please check the comments in #128 regarding acceptance tests. I've rebased this on the current upstream master.

File: openstack/sharedfilesystems/v2/shares/requests.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+)
+
+// CreateOptsBuilder allows extensions to add additional parameters to the
+// Create request.
+type CreateOptsBuilder interface {
+	ToShareCreateMap() (map[string]interface{}, error)
+}
+
+// CreateOpts used for ...
+type CreateOpts struct {
```
Comment (mvalkon) [inline openstack/sharedfilesystems/v2/shares/requests.go]: Done!

File: openstack/sharedfilesystems/v2/shares/requests.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	ShareProto         string            `json:"share_proto"`
```
Comment (mvalkon) [inline openstack/sharedfilesystems/v2/shares/requests.go]: Done

File: openstack/sharedfilesystems/v2/shares/requests.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	Size               int               `json:"size"`
```
Comment (mvalkon) [inline openstack/sharedfilesystems/v2/shares/requests.go]: Done

File: openstack/sharedfilesystems/v2/shares/requests.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+	Name               string            `json:"name,omitempty"`
+	Description        string            `json:"description,omitempty"`
+	DisplayName        string            `json:"display_name,omitempty"`
+	DisplayDescription string            `json:"display_description,omitempty"`
+	ShareType          string            `json:"share_type,omitempty"`
+	VolumeType         string            `json:"volume_type,omitempty"`
+	SnapshotID         string            `json:"snapshot_id,omitempty"`
+	IsPublic           bool              `json:"is_public,omitempty"`
```
Comment (mvalkon) [inline openstack/sharedfilesystems/v2/shares/requests.go]: Done

File: openstack/sharedfilesystems/v2/shares/requests.go [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+	MetaData           map[string]string `json:"metadata,omitempty"`
```
Comment (mvalkon) [inline openstack/sharedfilesystems/v2/shares/requests.go]: Done

File: openstack/sharedfilesystems/v2/shares/results.go [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+	// The host name of the share
+	Host string `json:"host"`
+	// The UUID of the share
+	ID string `json:"id"`
+	// Indicates the visibility of the share
+	IsPublic bool `json:"is_public,omitempty"`
+	// Share links for pagination
+	Links []map[string]string `json:"links"`
+	// Key, value -pairs of custom metadata
+	MetaData map[string]string `json:"metadata,omitempty"`
```
Comment (mvalkon) [inline openstack/sharedfilesystems/v2/shares/results.go]: Done

Comment (mvalkon) [inline openstack/sharedfilesystems/v2/shares/results.go]: Argh, so sorry! I overlooked this for some reason. Should be fixed now.

Comment (jrperritt) [inline openstack/sharedfilesystems/v2/shares/results.go]: it's quite alright

**Comment by jrperritt** [comment]
+2

**Pull request merged**